system/jlib/jfile.cpp

```diff
@@ -7106,3 +7109,230 @@ const FileSystemProperties & queryFileSystemProperties(const char * filename)
     else
         return linuxFileSystemProperties;
 }
+
+
+#ifndef _WIN32
+class CFileEventWatcher : public CInterfaceOf<IFileEventWatcher>, implements IThreaded
+{
+    class CMonitorItem : public CInterface
+    {
+    public:
+        CMonitorItem(const char *_filename, int _watchFd, FileWatchEvents _events) : filename(_filename), watchFd(_watchFd), events(_events)
+        {
+        }
+        std::string filename;
+        int watchFd = -1;
+        FileWatchEvents events = FileWatchEvents::none;
+        bool pendingRemoval = false;
+    };
+    int inotifyFd = -1;
+    int pipefd[2] = {-1, -1};
+    pollfd toPoll[2];
+    std::vector<Owned<CMonitorItem>> monitoredFiles;
```

Comment:
Is this what we are doing instead of using jarray these days?

Reply:
yes in general, for an owned arrays, and maps, tending to use stl.
More flexible, easier to use, and at least for maps, e.g. unordered_map, a fair bit faster if I remember correctly, although not massively time critical in this context.